controllers/clusterenvironment_controller.go

```diff
@@ -278,6 +278,20 @@ func (r *ClusterEnvironmentReconciler) MonitorHealth(ctx context.Context, ns str
 					}
 				}
 			}
+			if cfg == nil {
```

Comment:
we may handle this check in `clustercontext.GetClusterRESTConfig` and return an error from there. WDYT?

Reply:
There are other places where this function is called where empty config is compatible. So, I would prefer to retain the current code. I will work on the other suggestions.

Reply:
Right, good point!